objstore.go

```diff
@@ -442,7 +441,11 @@ func DownloadFile(ctx context.Context, logger log.Logger, bkt BucketReader, src,
 }
 
 // DownloadDir downloads all object found in the directory into the local directory.
-func DownloadDir(ctx context.Context, logger log.Logger, bkt BucketReader, originalSrc, src, dst string, options ...DownloadOption) error {
+func DownloadDir(ctx context.Context, logger log.Logger, bkt BucketReader, src, dst string, options ...DownloadOption) error {
+	return downloadDir(ctx, logger, bkt, src, src, dst, options...)
+}
+
+func downloadDir(ctx context.Context, logger log.Logger, bkt BucketReader, originalSrc, src, dst string, options ...DownloadOption) error {
```

Comment:
I'm wondering first of all whether `downloadDir` shouldn't return immediately if `ctx.Err()` is non-nil. The reason is that `downloadDir` can be called recursively with a context from `errgroup.WithContext`, that gets canceled when one or more of the goroutines fail.